docker-compose.yml

```diff
@@ -0,0 +1,6 @@
+version: "3.8"
+services:
+  mote:
+    build: .
+    ports:
+      - "5000:5000"
```

Comment:
I would suggest mounting the volume here in `docker-compose.yml` because, it'll keep track of the changes and make sure to rebuild the container in case of any modifications inside the volume.

When we build the container image by COPYing the volume inside Dockerfile, It actually builds into the Docker image with some sort of compression as a Volume specific to that image, resulting in manually rebuilding it when modifying the directory. Adding it to Docker-compose would make sure for us to not do it and makes the development experience much smoother.

(We can do the same with mounting it on the CLI when running it without Docker-compose)

```
volumes:
      - .:/mote
```

Reply:
> ```
> volumes:
>       - .:/mote
> ```

Understood. This is to mount the `/mote` folder as root of the container image.

> When we build the container image by COPYing the volume inside Dockerfile, It actually builds into the Docker image with some sort of compression as a Volume specific to that image, resulting in manually rebuilding it when modifying the directory. Adding it to Docker-compose would make sure for us to not do it and makes the development experience much smoother.

All points noted. But I have two questions regarding these suggested changes.
1. How do we plan on synchronizing the stuff with respect to the changes made to the local repo?
2. Do we rebuild the volume every time to constitute the changes are which are made?

Building an image with mounted volumes takes longer than simply mounting folders as volumes but it is sure to make things consistent and clean.

Reply:
> Understood. This is to mount the /mote folder as root of the container image.

It'll mount the current folder (in which the Docker-compose file lives) on `/mote` of the container Image.

> How do we plan on synchronizing the stuff with respect to the changes made to the local repo?

In a development environment, whenever there'll be changes to the code base, docker-compose will update it inside the container too. (Currently, Not the case as the codebase gets built in the Container image.)

> Do we rebuild the volume every time to constitute the changes are which are made?

We wouldn't have to, Docker-compose will take care of the volume management.

> Building an image with mounted volumes takes longer than simply mounting folders as volumes but it is sure to make things consistent and clean.

as we're using docker-compose, I think, It'll be a good idea to use features that it provides because, it'll ensure, whenever local changes are made, the user wouldn't have to interact with the Docker / Docker-compose CLI interface to rebuild the container on file changes. When the folder as volume is mounted, It'll be updated with in the container too.

P.S: We're not creating a separate volume but asking `docker-compose` to mount the current directory (in which `docker-compose.yml` lives) as a volume with in the container. 

What do you think ?

Reply:
That makes sense. I must have misinterpreted stuff previously and so were the questions, as I thought that for every small changes to be reflected in here - we would have to rebuild the image, which is clearly not the case.